hermes-common/src/main/java/pl/allegro/tech/hermes/common/admin/zookeeper/ZookeeperAdminCache.java

```diff
@@ -22,35 +24,44 @@
     private final ObjectMapper objectMapper;
     private final List<AdminOperationsCallback> adminCallbacks = new ArrayList<>();
 
+    private long initializationTime;
+
     @Inject
-    public ZookeeperAdminCache(ZookeeperPaths zookeeperPaths, @Named(CuratorType.HERMES) CuratorFramework client, ObjectMapper objectMapper) {
+    public ZookeeperAdminCache(ZookeeperPaths zookeeperPaths,
+                               @Named(CuratorType.HERMES) CuratorFramework client,
+                               ObjectMapper objectMapper,
+                               Clock clock) {
         super(client, zookeeperPaths.adminPath(), true);
         this.objectMapper = objectMapper;
-
+        this.initializationTime = clock.millis();
         getListenable().addListener(this);
     }
 
     @Override
     public void childEvent(CuratorFramework client, PathChildrenCacheEvent event) throws Exception {
         switch (event.getType()) {
+            case CHILD_UPDATED:
             case CHILD_ADDED:
-                if (event.getData().getPath().contains(RETRANSMIT.name())) {
-                    retransmit(client, event);
+                if (event.getData().getPath().contains(RETRANSMIT.name()) && isYoungerThanThisNode(event)) {
+                    retransmit(event);
                 }
                 break;
             default:
                 break;
         }
     }
 
-    private void retransmit(CuratorFramework client, PathChildrenCacheEvent event) throws Exception {
+    private boolean isYoungerThanThisNode(PathChildrenCacheEvent event) {
+        Stat stat = event.getData().getStat();
+        return stat.getCtime() > initializationTime && stat.getMtime() > initializationTime;
```

Comment:
Yeah, `mtime` is sufficient anyway because `mtime` is at least as large as `ctime`.